File: assets/codebase/menu/MenuItemComponent.cs
namespace city.menu {
    /// <summary>
    /// Stores the serialized runtime metadata required to navigate one baked city demo menu item.
    /// </summary>
    public sealed class MenuItemComponent : Component {
        /// <summary>
        /// Current payload version used by scene persistence for baked menu items.
        /// </summary>
        public const byte CurrentVersion = 1;

        /// <summary>
        /// Stable serialized component type id used by baked menu item scene records.
        /// </summary>
        public const string SerializedComponentTypeId = "city.menu.MenuItemComponent, gameplay";

        /// <summary>
        /// Backing field for the owning panel id.
        /// </summary>
        string PanelIdValue;

        /// <summary>
        /// Backing field for the stable item id.
        /// </summary>
        string ItemIdValue;

        /// <summary>
        /// Backing field for the item description rendered when selected.
        /// </summary>
        string DescriptionValue;

        /// <summary>
        /// Backing field for the action target id or scene path.
        /// </summary>
        string TargetIdValue;

        /// <summary>
        /// Initializes a new baked city menu item component with empty metadata.
        /// </summary>
        public MenuItemComponent() {
            PanelIdValue = string.Empty;
            ItemIdValue = string.Empty;
            DescriptionValue = string.Empty;
            TargetIdValue = string.Empty;
        }

        /// <summary>
        /// Gets or sets the stable panel id that owns the item.
        /// </summary>
        public string PanelId {
            get { return PanelIdValue; }
            set { PanelIdValue = value ?? string.Empty; }
        }

        /// <summary>
        /// Gets or sets the stable item id used for selection tracking.
        /// </summary>
        public string ItemId {
            get { return ItemIdValue; }
            set { ItemIdValue = value ?? string.Empty; }
        }

        /// <summary>
        /// Gets or sets the supporting description shown when the item is selected.
        /// </summary>
        public string Description {
            get { return DescriptionValue; }
            set { DescriptionValue = value ?? string.Empty; }
        }

        /// <summary>
        /// Gets or sets the action kind performed when the item is confirmed.
        /// </summary>
        public MenuActionKind ActionKind { get; set; }

        /// <summary>
        /// Gets or sets the action target id or scene path.
        /// </summary>
        public string TargetId {
            get { return TargetIdValue; }
            set { TargetIdValue = value ?? string.Empty; }
        }

        /// <summary>
        /// Gets or sets the idle fill color applied to the baked row background.
        /// </summary>
        public byte4 IdleFillColor { get; set; }

        /// <summary>
        /// Gets or sets the idle border color applied to the baked row background.
        /// </summary>
        public byte4 IdleBorderColor { get; set; }

        /// <summary>
        /// Gets or sets the selected fill color applied to the baked row background.
        /// </summary>
        public byte4 SelectedFillColor { get; set; }

        /// <summary>
        /// Gets or sets the selected border color applied to the baked row background.
        /// </summary>
        public byte4 SelectedBorderColor { get; set; }
    }
}

File: windows-build-20260528/cooked/code/gameplay/MenuItemComponent.cpp
#ifdef DrawText
#undef DrawText
#endif
#include "MenuItemComponent.hpp"
#include "runtime/native_string.hpp"
#include "MenuActionKind.hpp"
#include "byte4.hpp"
#include "runtime/native_string.hpp"

uint8_t MenuItemComponent::CurrentVersion = 1;

std::string MenuItemComponent::SerializedComponentTypeId = "city.menu.MenuItemComponent, gameplay";

::MenuActionKind MenuItemComponent::get_ActionKind()
{
return this->ActionKind;
}

void MenuItemComponent::set_ActionKind(::MenuActionKind value)
{
this->ActionKind = value;
}

byte4 MenuItemComponent::get_IdleBorderColor()
{
return this->IdleBorderColor;
}

void MenuItemComponent::set_IdleBorderColor(byte4 value)
{
this->IdleBorderColor = value;
}

byte4 MenuItemComponent::get_IdleFillColor()
{
return this->IdleFillColor;
}

void MenuItemComponent::set_IdleFillColor(byte4 value)
{
this->IdleFillColor = value;
}

const std::string& MenuItemComponent::get_ItemId()
{
return this->ItemIdValue;}

void MenuItemComponent::set_ItemId(std::string value)
{
this->ItemIdValue = value;
}

const std::string& MenuItemComponent::get_PanelId()
{
return this->PanelIdValue;}

void MenuItemComponent::set_PanelId(std::string value)
{
this->PanelIdValue = value;
}

byte4 MenuItemComponent::get_SelectedBorderColor()
{
return this->SelectedBorderColor;
}

void MenuItemComponent::set_SelectedBorderColor(byte4 value)
{
this->SelectedBorderColor = value;
}

byte4 MenuItemComponent::get_SelectedFillColor()
{
return this->SelectedFillColor;
}

void MenuItemComponent::set_SelectedFillColor(byte4 value)
{
this->SelectedFillColor = value;
}

const std::string& MenuItemComponent::get_TargetId()
{
return this->TargetIdValue;}

void MenuItemComponent::set_TargetId(std::string value)
{
this->TargetIdValue = value;
}

MenuItemComponent::MenuItemComponent() : ActionKind(), IdleBorderColor(), IdleFillColor(), SelectedBorderColor(), SelectedFillColor(), ItemIdValue(), PanelIdValue(), TargetIdValue()
{
this->PanelIdValue = String::Empty;
this->ItemIdValue = String::Empty;
this->TargetIdValue = String::Empty;
}

